cmd/snap-discard-ns/snap-discard-ns.c

```diff
@@ -41,22 +42,61 @@
 #endif
 
 int main(int argc, char** argv) {
-    if (argc != 2) {
-        printf("Usage: snap-discard-ns <SNAP-INSTANCE-NAME>\n");
-        return 0;
+    bool from_snap_confine = false;
```

Comment:
Sorry for the bike shed but this parsing feels like a bit more than we need. Unless we know this will grow more options and positional arguments - otherwise I would say YAGNI and go with something shoter like:
```
int posArgIndex = 1;
    if (argc > 3) {
       printf("error: too many arguments");
       return 1;
    }
    if (argc == 3) {
       if (!sc_streq(argv[2], "--from-snap-confine")) {
          printf("error: incorrect option\n");
          return 1;
       }
       posArgIndex = 2;
    }
    const char* snap_instance_name = argv[posArgIndex];
    struct sc_error* err = NULL;
    sc_instance_name_validate(snap_instance_name, &err);
    sc_die_on_error(err);
```

Reply:
Eventually I want to use one reusable parser from libsnap-confine-private, one that is compliant with some usability standards and has full tests. Looking at the suggestion I can apply it though it's not terribly shorter so not sure if really needed.